### ISSUE TITLE
Tooling for convergence comparison needed

### ISSUE THREAD

**Issue by raver119** [deeplearning4j/deeplearning4j#3398]
Since @crockpotveggies got some issues with convergence time with ParallelWrapper, and we're going to introduce multiple different asynchronous updaters for parallel training, it would be really nice to have something that would allow basic comparison of learning process efficiency.

I.e. we could introduce TrainingListener that'll take scores and gradients at specified frequency, and build virtual curves like we have in UI, but without UI, comparison should be made automatically, without human interference.

**Comment by crockpotveggies** [comment]
As discussed in Gitter, we can convert the training curve into a polygon and use Java Area to determine the total area under training curves for different training instances. We can use this as a measurement to determine if parallel training converges faster (or slower) than single GPU.

**Comment by raver119** [comment]
Yes, + we can compare skewness of different curves, bifurcation points etc. So basically we should have nice tooling with minimal efforts, that'll allow us to compare different training setups.

**Comment by AlexDBlack** [comment]
This sounds good to me.
I'd be interested in obtaining both performance versus number of iterations (total minibatches processed by all as well as by one machine), and performance versus time charts.
Other statistics, such as the amount of time spent during averaging, ETL etc would also be good.

I don't imagine this would be especially hard to produce. We just have to be careful in how we collect times so they are not biased by the evaluation procedure.

**Comment by raver119** [comment]
@crockpotveggies what's up there?
